docs/build/use-cases/trustless-agent.mdx

```diff
@@ -0,0 +1,308 @@
+  const resp = await makeAppdRequest("POST", "/rofl/v1/keys/generate", {
+    key_id: keyId,
+    kind: "secp256k1",
+  });
+  return resp; // contains { key: "<hex_without_0x>" } inside enclave
+}
+
+export async function submitEthTx(to, data, gasLimit = 200000) {
+  // Ask appd to sign+submit a Sapphire EVM tx using the enclave key.
+  const payload = {
+    tx: {
+      kind: "eth",
+      data: { gas_limit: gasLimit, to, value: 0, data },
+    },
+  };
+  const resp = await makeAppdRequest("POST", "/rofl/v1/tx/sign-submit", payload);
+  return resp;
+}
+
+export function encodeCall(abi, method, args) {
+  const iface = new Interface(abi);
+  return iface.encodeFunctionData(method, args);
+}
+```
+</details>
+
+<details>
+  <summary>docker/Dockerfile</summary>
+
+```dockerfile
+FROM oven/bun:1.1-alpine
+WORKDIR /app
+
+# Copy your ElizaOS project (created with `elizaos create ...`).
+ADD . /app
+
+# Install ElizaOS CLI and project dependencies.
+RUN bun install -g @elizaos/cli && bun install
+
+# Minimal entrypoint that launches the agent.
+ENTRYPOINT ["/bin/sh", "/app/docker/entry.sh"]
+```
+</details>
+
+### Compose
+
+Add `compose.yaml` at project root:
+
+<details>
+  <summary>compose.yaml</summary>
+
+```yaml
+services:
+  eliza-agent:
+    build: ./docker
+    image: docker.io/YOUR_USERNAME/rofl-eliza:latest
+    platform: linux/amd64
+    environment:
+      - OPENAI_API_KEY=${OPENAI_API_KEY}
+    volumes:
+      # Mount only if your agent needs to interact with Sapphire via appd.
+      # - /run/rofl-appd.sock:/run/rofl-appd.sock
+```
+</details>
+
+Build and push:
+
+```shell
+docker compose build
+docker compose push
+```
+
+For extra security, **pin the image digest** in `compose.yaml`.
+
+## Contract interaction (high level)
+
+Your agent can call Sapphire contracts to log decisions or perform actions.
+Gate privileged paths in your contracts by verifying ROFL origin.
+
+## Build the ROFL bundle
+
+<Tabs>
+  <TabItem value="Native Linux">
+
+```shell
+oasis rofl build
+```
+
+  </TabItem>
+  <TabItem value="Docker (Windows, Mac, Linux)">
+
+```shell
+docker run --platform linux/amd64 --volume .:/src -it \
+  ghcr.io/oasisprotocol/rofl-dev:main oasis rofl build
+```
+
+  </TabItem>
+</Tabs>
+
+This produces the `.orc` bundle for deployment, sealing in enclave identity.
+
+## Update on-chain app config
+
+```shell
+oasis rofl update
+```
+
+## Deploy to a ROFL provider
+
+```shell
+oasis rofl deploy
+```
+
+By default, the Oasis-maintained provider is selected on Testnet.
+
+## How the agent interacts on Sapphire
+
+- Encode calldata for your contract method and call `submitEthTx(...)` via
+  `appd`. The tx originates from the enclave-bound key of your ROFL app.
+- Contracts can require verified ROFL origin to bind calls to your app.
+
+## Troubleshooting
+
+- Ensure the ROFL machine can resolve your model endpoint (OpenAI/Ollama)
+- Check `oasis rofl machine logs` for enclave startup issues
```

Comment:
```suggestion
- Check `oasis rofl machine logs` for enclave startup issues
```

I would combine the Troubleshooting, Resources and Conclusion in something like "Testing it out" section where you'd show how to use `oasis rofl machine logs` and what to expect in the output.